plugins/steelseries/fu-steelseries-fizz-hid.c

```diff
@@ -28,19 +28,68 @@ static gboolean
 fu_steelseries_fizz_hid_command(FuDevice *device, guint8 *data, gsize datasz, GError **error)
 {
 	gboolean ret;
-
-	ret = fu_udev_device_pwrite(FU_UDEV_DEVICE(device), 0, data, datasz, error);
-	if (!ret) {
-		g_prefix_error(error, "failed to write report: ");
-		return FALSE;
+	guint8 report_id = 0;
+	guint8 rdata[STEELSERIES_BUFFER_REPORT_SIZE] = {0};
+
+	/* In BT mode the sync and data channels are sharing the device descriptor
+	 * with the management channel. This is the reason why we receive
+	 * "unexpected" packets with 0x01 or 0x05 Report IDs over the same
+	 * descriptor on mouse connecting, waking up or just moving the mouse.
+	 * */
+	while (TRUE) {
+		g_autoptr(GError) pread_error = NULL;
+
+		/* Force the request for each iteration to avoid the
+		 * situation with an infinite loop due the lost single packet.
+		 * This is safe since the device doesn't support the update over BT.
+		 * */
+		ret = fu_udev_device_pwrite(FU_UDEV_DEVICE(device), 0, data, datasz, error);
+		if (!ret) {
```

Comment:
no need for ret

Reply:
why? what if it fails?

Reply:
Ohh I meant `if (!fu_udev_device_pwrite(FU_UDEV_DEVICE(device), 0, data, datasz, error)) {`